cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +83,7 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, MapperInput> inputs;
```

Comment:
further down it seems that it is the class name of the input format provider. But you use the same class for all dataset inputs. That means, multiple dataset inputs will cancel out each other when added to this map?